std-bits/base/src/main/java/org/enso/base/enso_cloud/HideableValue.java

```diff
@@ -0,0 +1,33 @@
+package org.enso.base.enso_cloud;
+
+/** Represents a value that is input of various operation that may contain a Secret. */
+public sealed interface HideableValue permits HideableValue.SecretValue, HideableValue.PlainValue {
+
+  record SecretValue(String secretId) implements HideableValue {
+    @Override
+    public String render() {
+      return "__SECRET__";
+    }
+
+    @Override
+    public String safeResolve() throws EnsoSecretAccessDenied {
+      throw new EnsoSecretAccessDenied();
+    }
+  }
+
+  record PlainValue(String value) implements HideableValue {
```

Comment:
Secure computing suggests to store values in `byte[]` - because (unlike String) `byte[]` can be programaticaly destroyed. That gives the user of the `HideableValue` control of the life-span a value is valid.

Reply:
The `PlainValue` is nothing that has to be destroyed.

This PR is not changing _any_ of the 'secure' secret logic from #8006, it just refactors it a bit and changes the API.

Also note that for the actual secrets (not something the `HideableValue` _is_), we have to hold them as `String` at some point because we need to pass it to places like JDBC properties.

---

I think the comment suggests that `HideableValue` is a misleading name. I really didn't have better ideas how to call this type. Any suggestions?

`HideableValue` is a type that can either be plain non-sensitive text value OR a secret **id** that will be _resolved into a proper sensitive value much later_ (the ID itself is nothing that has to be secure).